### DISCUSSION TITLE
morphia upgrade

### DISCUSSION THREAD

**Discussion by emipascal** [SDA-SE/sda-dropwizard-commons#1827] [Q&A]
is there a plan for upgrading morphia from 1.6 to latest version?

**Comment by christopher-cudennec** [comment]
Hello @emipascal !

I think I need to check my notification settings. Sorry for the long delay! 🌻 

There is a long story to tell behind the Morphia upgrade. Summary: We will drop Morphia and add Spring Data Mongo as technology for accessing MongoDB. We're currently working on it for version 3 of sda-dropwizard-commons. See #1469 

More background information:
* we want version 3 to finally support Java 17 😃 
* Morphia is really slow on providing new versions and has been the reasons why we could not upgrade the Java version of sda-dropwizard-commons. When trying to take over Morphia 2 we ran into major breaking changes because Morphia dropped several features that we needed for the upgrade. In the end we decided against Morphia and for Spring Data Mongo because it is much better supported and really feature rich. It might also be a good migration path if you plan to switch from Dropwizard to Spring. You might even take a look at [sda-spring-boot-commons](https://github.com/SDA-SE/sda-spring-boot-commons).

I hope you find the information helpful.

Cheers,

Christopher
